firebase-firestore/src/main/java/com/google/firebase/firestore/remote/RemoteStore.java

```diff
@@ -162,6 +169,23 @@ public RemoteStore(
     onlineStateTracker =
         new OnlineStateTracker(workerQueue, remoteStoreCallback::handleOnlineStateChange);
 
+    new Thread() {
+      public void run() {
+        try {
+          ProviderInstaller.installIfNeeded(context);
+        } catch (GooglePlayServicesNotAvailableException /* Thrown by ProviderInstaller */
+            | GooglePlayServicesRepairableException /* Thrown by ProviderInstaller */
+            | IllegalStateException /* Thrown by Robolectric */ e) {
```

Comment:
Note: Apparently there is a way to avoid this IllegalStateException by setting "com.google.android.gms.version" in the AndroidManifest that Robolectric uses, but this seems to be fully automated in our repo (and should be as of Robolectric 4) and I can't figure out how to do this.